core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java

```diff
@@ -43,6 +48,7 @@ public ReflectiveHierarchyIgnoreWarningBuildItem ignoreJavaClassWarnings() {
 
     @BuildStep
     public void build(CombinedIndexBuildItem combinedIndexBuildItem,
+            List<GeneratedClassesIndexBuildItem> generatedClassesIndexBuildItem,
```

Comment:
I wonder if we should just have an index build item including everything. It's not the first time we have this issue and we are building combined indexes in a few places and also we often forget about it (you fixed it in a couple of places already).

I think the main issue we have with this is we would need to find a name that is clear enough.

Reply:
the main issue is not really the name but where it should reside... as I mention in the first comment about moving the BeanArchiveIndexBuildItem to core but that would mean quite some changes in all extensions so having this is more flexible in my opinion as whatever extension needs to provide their generated classes can do it easily.

So it would be ideal to have one build item that has access to all of this but since there are many thing that consume such info and it is spread across various levels - core vs extensions this approach looked the least intrusive.